### PR TITLE
Support Multiple DNS period resolving mechanism

### LINKED ISSUE #6222: dns resolve for skywalking trace receiver service addresses
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [X ] Question or discussion
- [ ] Bug
- [ ] Requirement
- [ ] Feature or performance improvement

___
### Question
- What do you want to know?
I want to add dns resolver for the domain of  skywalking trace receiver service
-  WHY ?
   - It's hard to maintain agent config
`collector.backend_service=${SW_AGENT_COLLECTOR_BACKEND_SERVICES:127.0.0.1:11800}
`
   - If extending OAP service, we must restart all service.
  

- HOW ?
  - In this situation, I want to set domain to this variable.The domain has a number of A records
  - modify the method boot() of GRPCChannelManager, check whether it is domain, if so ,then resolve it ,and add all ip addresses to serverList
```
 public void boot() {
        if (Config.Collector.BACKEND_SERVICE.trim().length() == 0) {
            LOGGER.error("Collector server addresses are not set.");
            LOGGER.error("Agent will not uplink any data.");
            return;
        }
        grpcServers = Arrays.asList(Config.Collector.BACKEND_SERVICE.split(","));
        // resolve all domain to ip and add to server List
        connectCheckFuture = Executors.newSingleThreadScheduledExecutor(
            new DefaultNamedThreadFactory("GRPCChannelManager")
        ).scheduleAtFixedRate(
            new RunnableWithExceptionProtection(
                this,
                t -> LOGGER.error("unexpected exception.", t)
            ), 0, Config.Collector.GRPC_CHANNEL_CHECK_INTERVAL, TimeUnit.SECONDS
        );
    }

```
- I know than nginx can help me .But since skywalking already have client loadbalance, I don't want to use nginx, because it could bring some complexity

### PR DESCRIPTION
**Pull request by ThisSeanZhang**
The previous code only supports DNS reverse check of the first domain config
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #6222.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/remote/GRPCChannelManager.java [hunk 1]
```diff
@@ -99,20 +101,28 @@ public void shutdown() {
...
-                grpcServers = newGrpcServers;
-            } catch (Throwable t) {
-                LOGGER.error(t, "Failed to resolve {} of backend service.", backendService);
-            }
+            grpcServers = Arrays.stream(Config.Collector.BACKEND_SERVICE.split(","))
+                    .filter(StringUtil::isNotBlank)
+                    .map(eachBackendService -> eachBackendService.split(":"))
+                    .filter(domainPortPairs -> {
+                        if (domainPortPairs.length < 2) {
+                            LOGGER.debug(new IllegalArgumentException(), "Service address [{}] format error", domainPortPairs[0]);
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/remote/GRPCChannelManager.java:109]: Don't log debug with a new exception, this is pointless.

File: CHANGES.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ Release Notes.
 #### Project
 
 #### Java Agent
-
+* Support Multiple DNS period resolving mechanism
 #### OAP-Backend
```
Comment (wu-sheng) [inline CHANGES.md:12]:
```suggestion
* Support Multiple DNS period resolving mechanism

#### OAP-Backend
```

**Review by wu-sheng — COMMENTED** [review]
Besides the code level review, what is the case of multiple DNS addresses?

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/remote/GRPCChannelManager.java (hunk 1 above)
Comment (wu-sheng) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/remote/GRPCChannelManager.java:109]:
```suggestion
                            LOGGER.debug("Service address [{}] format error. The expected format is IP:port", domainPortPairs[0]);
```

**Comment by ThisSeanZhang** [comment]
Avoid inertial thinking configuration,When Config DNS. Or Same IP but diff Port.

**Review by wu-sheng — APPROVED** [review]

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**